lib/api/CFieldDataCategorizer.cc

```diff
@@ -425,8 +425,8 @@ bool CFieldDataCategorizer::periodicPersistStateInBackground() {
                       // Do NOT add std::ref wrappers
                       // around these arguments - they
                       // MUST be copied for thread safety
-                      m_DataCategorizer->makePersistFunc(), m_ExamplesCollector,
-                      std::placeholders::_1)) == false) {
+                      m_DataCategorizer->makeBackgroundPersistFunc(),
```

Comment:
Testing note: if `makeBackgroundPersistFunc` is changed to `makeForegroundPersistFunc` in this line then the newly added unit test fails a very high proportion of the time.  This proves that the fix was necessary and the fact that the test always passes with `makeBackgroundPersistFunc` suggests the fix works.